src/sentry/migrations/0166_create_notificationsetting_table.py

```diff
@@ -0,0 +1,166 @@
+# Generated by Django 1.11.29 on 2021-02-25 20:08
+
+from django.db import migrations
+import sentry.db.models.fields.bounded
+import sentry.models.notificationsetting
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0165_metric_alerts_fix_group_ids"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="NotificationSetting",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                (
+                    "scope_type",
+                    sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
```

Comment:
Are we concerned about storage space here? Using strings will make this table easier for humans to use both locally and in adhoc query tools.

Reply:
Our team has decided to use enumerations in the DB whenever possible. We want to transition to [enumerated types](https://www.postgresql.org/docs/9.1/datatype-enum.html) as soon as they become available.

Reply:
I'd suggest not using postgres enums, since they require a db migration whenever you want to change them. Ideally we'd just have a Django level `Enum` column that translates to an int column.

Reply:
Enumerated types give you textual representations to work with though. Storing integers will make writing adhoc queries and working with the models in a repl more difficult as you'll have to remember all the numeric value mappings.

When you're ready to switch to enums wouldn't both a char and integer column require a multi-step migration to change the column type over?